utils/comparable_bytes.cc

```diff
@@ -23,6 +25,120 @@ static void convert_signed_fixed_length_integer(managed_bytes_view& src, bytes_o
+    size_t num_bits = (8 - leading_sign_bits_in_curr_byte) + src.size_bytes() * 8;
+    // Calculate the number of length bits (inverted sign bits in the prefix)
+    // that indicate the total size of the encoded format in bytes.
+    // 0-6 bits => 1 byte; 7-13 => 2 bytes; etc to 56-63 => 9 bytes.
+    // (63 bits will need 10 bytes, but this encoding does it with 9)
+    size_t num_length_bits_to_write = std::min(size_t(9), num_bits / 7 + 1);
+
+    if (num_length_bits_to_write >= 8) {
+        // Write one byte full of length bits (i.e. inverted sign bits)
+        out.write<uint8_t>(~sign_only_byte);
+        num_length_bits_to_write -= 8;
+    }
+
+    if (num_length_bits_to_write > 0) {
+        if (leading_sign_bits_in_curr_byte > num_length_bits_to_write || num_bits == 63) {
+            // The remaining length bits can be safely incorporated into curr_byte by
+            // inverting the existing sign bits while keeping at least one sign bit intact.
+            // Note that num_bits == 63, is a special case where the only remaining sign bit is flipped but
+            // this doesn't corrupt the actual value, as the decoding logic accounts for this scenario.
+            curr_byte ^= (int8_t(BYTE_SIGN_MASK) >> (num_length_bits_to_write - 1));
+        } else {
+            // Insufficient number of sign bits in curr_byte.
+            // Write the remaining length bits as a separate byte.
+            out.write<uint8_t>(int8_t(sign_only_byte ^ BYTE_SIGN_MASK) >> (num_length_bits_to_write - 1));
+        }
+    }
+
+    // Write the curr_byte and rest of src
+    out.write<uint8_t>(curr_byte);
+    out.write(src);
+}
+
+// Refer encode_signed_long_type() for the encoding details
+static void decode_signed_long_type(managed_bytes_view& src, bytes_ostream& out) {
+    uint8_t curr_byte = read_simple<uint8_t>(src);
+    uint8_t length_bit = curr_byte & BYTE_SIGN_MASK;
+    // Deduce bytes to read from the length bits (i.e inverted sign bits) in curr_byte
+    uint8_t length_bits_in_curr_byte = length_bit ? std::countl_one(curr_byte) : std::countl_zero(curr_byte);
+    uint8_t bytes_to_read = length_bits_in_curr_byte - 1; // -1 to account for the byte already read
+    if (bytes_to_read == 7) {
+        // check also the first bit in next byte for length bit
+        curr_byte = read_simple<uint8_t>(src);
+        length_bits_in_curr_byte = (curr_byte & BYTE_SIGN_MASK) == length_bit ? 1 : 0;
+        if (length_bits_in_curr_byte == 0) {
+            // not a length bit but a byte has been read
+            bytes_to_read--;
+        }
+    }
+
+    // Fill all leading bytes with sign bits, leaving space for the remaining bytes in src and curr_byte.
+    uint8_t bytes_with_sign_bit = sizeof(int64_t) - (1 + bytes_to_read);
+    const uint8_t sign_only_byte = int8_t(~length_bit) >> 7;
+    while (bytes_with_sign_bit--) {
+        out.write<uint8_t>(sign_only_byte);
```

Comment:
Again, collect the results into a scalar and serialize the entire scalar with one operation.

Reply:
Done.